lib/vm/opFns.js

```diff
@@ -587,7 +587,10 @@ module.exports = {
     }
 
     subMemUsage(runState, offset, length)
-    const data = runState.memory.read(offset.toNumber(), length.toNumber())
+    let data = Buffer.alloc(0)
+    if (!length.isZero()) {
```

Comment:
Why would it be a problem allowing reading 0 length?

Reply:
Checking for 0 length was being done in `memLoad`. I tried omitting it, but faced a problem: tests with an offset wider than 53 bits would fail (`BN.toNumber()` would throw). In these test cases the length is 0 (as even length 1 would need a lot of gas I think). So I found this check is basically a way to make the tests pass! I'm thinking about how we can improve on this.